RadixWallet/Features/AccountPreferencesFeature/Children/DeleteAccount/ChooseReceivingAccountOnDelete+Reducer.swift

```diff
@@ -0,0 +1,134 @@
+import ComposableArchitecture
+import Sargon
+import SwiftUI
+
+// MARK: - ChooseReceivingAccountOnDelete
+struct ChooseReceivingAccountOnDelete: Sendable, FeatureReducer {
+	struct State: Sendable, Hashable {
+		let accountToDelete: Account
+		var chooseAccounts: ChooseAccounts.State
+		var footerControlState: ControlState = .enabled
+
+		@PresentationState
+		var destination: Destination.State? = nil
+
+		init(accountToDelete: Account, chooseAccounts: ChooseAccounts.State) {
+			self.accountToDelete = accountToDelete
+			self.chooseAccounts = chooseAccounts
+		}
+	}
+
+	@CasePathable
+	enum ViewAction: Sendable, Equatable {
+		case continueButtonTapped([ChooseAccountsRow.State])
+		case skipButtonTapped
+	}
+
+	@CasePathable
+	enum ChildAction: Sendable, Equatable {
+		case chooseAccounts(ChooseAccounts.Action)
+	}
+
+	@CasePathable
+	enum DelegateAction: Sendable, Equatable {
+		case finished(AccountAddress?)
+	}
+
+	struct Destination: DestinationReducer {
+		@CasePathable
+		enum State: Sendable, Hashable {
+			case confirmSkipAlert(AlertState<Action.ConfirmSkipAlert>)
+			case tooManyAssetsAlert(AlertState<Action.TooManyAssetsAlert>)
```

Comment:
Yes, but I expect the `createDeleteAccountManifest` function from Sargon to perform the check and throw a specific error that hosts can handle. Would that be possible?

Reply:
yes, it is already done [here](https://github.com/radixdlt/sargon/pull/259/files#diff-97b4d7f3bfab301b14d8e78a84347912af6e09704cd36fe48c5f8f7ff89954c6R48-R53). However, the manifest won't be created here (but on `DeleteAccountConfirmationReducer`) so maybe the alert should be there?

Also, this triggers another comment: wouldn't it help to have a `DeleteAccountCoordinator` that is in charge of presenting/pushing views/alerts, as well as executing the logic to actually delete the account?

Reply:
I've added `DeleteAccountCoordinator`, which handles the delete account logic. I didn’t include a navigation stack since it made things unnecessarily complicated. Instead, I kept only `destination`, so `tooManyAssetsAlert` is presented by `ChooseReceivingAccountOnDelete`.